kernel/src/snapshot.rs

```diff
@@ -50,6 +53,11 @@ impl std::fmt::Debug for Snapshot {
 }
 
 impl Snapshot {
+    /// Create a new [`SnapshotBuilder`] to build a [`Snapshot`] for a given table root.
+    pub fn build(table_root: Url) -> SnapshotBuilder {
+        SnapshotBuilder::new(table_root)
+    }
```

Comment:
the `build().build_at(..)` feels very confusing. Why not just do `SnapshotBuilder::new().build_at(..)`?